### PR TITLE
Fix lightning implementation, docs and tests

### PR DESCRIPTION
**Pull request by NicolasDorier**

### CODE REVIEW

File: BTCPayServer.Client/Models/LightningNodeInformationData.cs [hunk 1]
```diff
@@ -9,7 +9,8 @@ namespace BTCPayServer.Client.Models
 {
     public class LightningNodeInformationData
     {
-        public IEnumerable<string> NodeInfoList { get; set; }
+        [JsonProperty("nodeURIs")]
+        public IEnumerable<string> NodeURIs { get; set; }
```
Comment (Kukks) [inline BTCPayServer.Client/Models/LightningNodeInformationData.cs:13]: Is this possible to ` IEnumerable<NodeInfo> ` and then decorate with `[JsonConverter(typeof(NodeUriJsonConverter))]`?

File: BTCPayServer.Client/Models/GreenfieldValidationError.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        {
+            if (path == null)
+                throw new ArgumentNullException(nameof(path));
+            if (message == null)
+                throw new ArgumentNullException(nameof(message));
+            Path = path;
+            Message = message;
+        }
+
+        public string Path { get; set; }
```
Comment (Kukks) [inline BTCPayServer.Client/Models/GreenfieldValidationError.cs:23]: Should we include `Code` here too or overkill?

File: BTCPayServer/wwwroot/swagger/v1/swagger.template.lightning.internal.json [hunk 3]
```diff
@@ -290,13 +254,14 @@
...
                             "application/json": {
                                 "schema": {
-                                    "type": "string"
+                                    "type": "string",
+                                    "description": "A bitcoin address belonging to the lightning node"
                                 }
                             }
                         }
                     },
-                    "403": {
```
Comment (Kukks) [inline BTCPayServer/wwwroot/swagger/v1/swagger.template.lightning.internal.json:298]: 403 is still possible when you have insufficient permissions on the api key

Comment (NicolasDorier) [reply BTCPayServer/wwwroot/swagger/v1/swagger.template.lightning.internal.json:298]: I think we should not mention it in the doc, as this is true for every endpoints.

File: BTCPayServer.Client/Models/LightningNodeInformationData.cs (hunk 1 above)
Comment (NicolasDorier) [reply BTCPayServer.Client/Models/LightningNodeInformationData.cs:13]: yes, will do

File: BTCPayServer.Client/Models/GreenfieldValidationError.cs (hunk 2 above)
Comment (NicolasDorier) [reply BTCPayServer.Client/Models/GreenfieldValidationError.cs:23]: overkill

**Pull request merged**